engine/consensus/compliance/engine.go

```diff
@@ -224,71 +236,56 @@ func (e *Engine) WithConsensus(hot module.HotStuff) *Engine {
 	return e
 }
 
-// Ready returns a ready channel that is closed once the engine has fully
-// started. For consensus engine, this is true once the underlying consensus
-// algorithm has started.
-func (e *Engine) Ready() <-chan struct{} {
+// Start starts the Hotstuff event processMessagesLoop, then the compliance engine worker threads.
+func (e *Engine) Start(ctx irrecoverable.SignalerContext) {
 	if e.core.hotstuff == nil {
-		panic("must initialize compliance engine with hotstuff engine")
+		ctx.Throw(fmt.Errorf("must initialize compliance engine with hotstuff engine"))
 	}
-	e.lm.OnStart(func() {
-		e.unit.Launch(e.loop)
-		e.unit.Launch(e.finalizationProcessingLoop)
-
-		ctx, cancel := context.WithCancel(context.Background())
-		signalerCtx, hotstuffErrChan := irrecoverable.WithSignaler(ctx)
-		e.stopHotstuff = cancel
 
-		// TODO: this workaround for handling fatal HotStuff errors is required only
-		//  because this engine and epochmgr do not use the Component pattern yet
-		e.unit.Launch(func() {
-			e.handleHotStuffError(hotstuffErrChan)
-		})
+	e.log.Info().Msg("starting hotstuff")
+	e.core.hotstuff.Start(ctx)
+	e.log.Info().Msg("hotstuff started")
 
-		e.core.hotstuff.Start(signalerCtx)
-		// wait for request handler to startup
+	e.log.Info().Msg("starting compliance engine")
+	e.Component.Start(ctx)
+	e.log.Info().Msg("compliance engine started")
+}
 
-		<-e.core.hotstuff.Ready()
-	})
-	return e.lm.Started()
+// Ready returns a ready channel that is closed once the engine has fully started.
+// For the consensus engine, we wait for hotstuff to start.
+func (e *Engine) Ready() <-chan struct{} {
+	// NOTE: this will create long-lived goroutines each time Ready is called
+	// Since Ready is called infrequently, that is OK. If the call frequency changes, change this code.
+	return util.AllReady(e.cm, e.core.hotstuff)
 }
 
 // Done returns a done channel that is closed once the engine has fully stopped.
 // For the consensus engine, we wait for hotstuff to finish.
 func (e *Engine) Done() <-chan struct{} {
-	e.lm.OnStop(func() {
-		e.log.Info().Msg("shutting down hotstuff eventloop")
-		e.stopHotstuff()
-		<-e.core.hotstuff.Done()
-		e.log.Info().Msg("all components have been shut down")
-		<-e.unit.Done()
-	})
-	return e.lm.Stopped()
-}
-
-// SubmitLocal submits an event originating on the local node.
-func (e *Engine) SubmitLocal(event interface{}) {
-	err := e.ProcessLocal(event)
-	if err != nil {
-		e.log.Fatal().Err(err).Msg("internal error processing event")
-	}
+	// NOTE: this will create long-lived goroutines each time Ready is called
+	// Since Ready is called infrequently, that is OK. If the call frequency changes, change this code.
```

Comment:
```suggestion
	// NOTE: this will create long-lived goroutines each time Done is called
	// Since Done is called infrequently, that is OK. If the call frequency changes, change this code.
```